apm-agent-attach/src/main/java/co/elastic/apm/attach/ElasticApmAttacher.java

```diff
@@ -151,14 +151,34 @@ public static void attach(String pid, Map<String, String> configuration, File ag
         File tempFile = createTempProperties(configuration, null);
         String agentArgs = tempFile == null ? null : TEMP_PROPERTIES_FILE_KEY + "=" + tempFile.getAbsolutePath();
 
-        ByteBuddyAgent.attach(agentJarFile, pid, agentArgs, ElasticAttachmentProvider.get());
+        attachWithFallback(agentJarFile, pid, agentArgs);
         if (tempFile != null) {
             if (!tempFile.delete()) {
                 tempFile.deleteOnExit();
             }
         }
     }
 
+    private static void attachWithFallback(File agentJarFile, String pid, String agentArgs) {
+        try {
+            // while the native providers may report to be supported and appear to work properly, in practice there are
+            // cases (Docker without '--init' option on some JDK images like 'openjdk:8-jdk-alpine') where the accessor
+            // returned by the provider will not work as expected at attachment time.
+            ByteBuddyAgent.attach(agentJarFile, pid, agentArgs, ElasticAttachmentProvider.get());
+        } catch (RuntimeException e1) {
+            try {
+                ByteBuddyAgent.attach(agentJarFile, pid, agentArgs, ElasticAttachmentProvider.getFallback());
+            } catch (RuntimeException e2) {
+                // output the two exceptions for debugging
+                System.err.println("Unable to attach with fallback provider:");
+                e2.printStackTrace();
+
+                System.err.println("Unable to attach with regular provider:");
+                e1.printStackTrace();
```

Comment:
[for reviewer] One major difference with the current implementation is that any failure during attachment won't prevent the application to start. While this might be desirable to fail more silently, it means that users are likely to not see any error unless they inspect their application standard output/logs.